lib/std/math.zig

```diff
@@ -406,6 +406,48 @@ pub const min3 = @compileError("deprecated; use @min instead");
 pub const max3 = @compileError("deprecated; use @max instead");
 pub const ln = @compileError("deprecated; use @log instead");
 
+/// Odd sawtooth function
+/// ```
+///         |
+///      /  | /    /
+///     /   |/    /
+///  --/----/----/--
+///   /    /|   /
+///  /    / |  /
+///         |
+/// ```
+/// Limit x to the half-open interval [-r, r).
+pub fn wrap(x: anytype, r: anytype) @TypeOf(x) {
+    return @mod(x + r, 2 * r) - r;
```

Comment:
Are `x` and `r` required to be signed? If so I would suggest adding a compile-time check for that.

Also, this function will overflow if `r` is more than half of its type's range e.g.:
```zig
test {
    const x: i8 = 120;
    const y: i8 = 100;
    _ = wrap(x, y);
}
```

Will overflow even though it is a meaningful expression with result `-80`

Reply:
you would check it by doing something like:
```zig
pub fn wrap(x: anytype, r: anytype) @TypeOf(x) {
     if (@typeInfo(@TypeOf(x)).Int.signedness == .signed or @typeInfo(@TypeOf(r)).Int.signedness == .signed) {
          @compileError("something");
     }

    return @mod(x + r, 2 * r) - r;
```